src/iop/clahe.c

```diff
@@ -99,9 +100,9 @@ void process(dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const void *c
     float *lm = luminance + (size_t)j * roi_out->width;
     for(int i = 0; i < roi_out->width; i++)
     {
-      double pmax = CLIP(fmax(in[0], fmax(in[1], in[2]))); // Max value in RGB set
-      double pmin = CLIP(fmin(in[0], fmin(in[1], in[2]))); // Min value in RGB set
-      *lm = (pmax + pmin) / 2.0;                           // Pixel luminocity
+      double pmax = CLIP(max3f(in)); // Max value in RGB set
```

Comment:
Should use the double variant as for  pmin

Reply:
Like this?
```c
  if(!dt_iop_have_required_input_format(4, self, piece->colors, ivoid, ovoid, roi_in, roi_out))
    return;

  dt_iop_rlce_data_t *data = piece->data;
  // pixel data is guaranteed to be 4-channel past the format check
  const float (*const __restrict__ in)[4] = (const float (*)[4])ivoid;
  const size_t num_pixels = (size_t)roi_out->width * roi_out->height;

  // PASS1: Get a luminance map of image...
  float *const __restrict__ luminance = dt_alloc_align_float(num_pixels);

  DT_OMP_FOR_SIMD()
  for(size_t i = 0; i < num_pixels; i++)
  {
    const float pmax = CLIP(max3f(in[i]));
    const float pmin = CLIP(min3f(in[i]));
    luminance[i] = (pmax + pmin) / 2.f; // Pixel luminosity
  }
  ```